securesystemslib/metadata.py

```diff
@@ -94,3 +90,70 @@ def pae(self) -> bytes:
             len(self.payload),
             self.payload,
         )
+
+    def sign(self, signer: Signer) -> Signature:
+        """Sign the payload and create the signature.
+
+        Arguments:
+            signer: A "Signer" class instance.
+
+        Returns:
+            A "Signature" instance.
+        """
+
+        signature = signer.sign(self.pae())
+        self.signatures.append(signature)
+
+        return signature
+
+    def verify(self, keys: List[Key], threshold: int) -> Dict[str, Key]:
+        """Verify the payload with the provided Keys.
+
+        Arguments:
+            keys: A list of "Key" class instance that is a public key to verify
+                the signatures.
```

Comment:
Minor nit:

```suggestion
            keys: A list of public keys to verify the signatures.
```
Should be good enough, right? The data type is shown by the type hint. 🤷